### PR TITLE
Add new emoji picker [SCI-4268]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-4268](https://biosistemika.atlassian.net/browse/SCI-4268)

### What was done
Add new emoji picker

### CODE REVIEW

File: app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js [hunk 1]
```diff
@@ -47,14 +48,56 @@ var RepositoryStatusColumnType = (function() {
...
-      events: {
-        emojibtn_click: function(button, event) {
-          validateForm();
-        }
+    var picker = new EmojiButton();
+    var icon = $(iconElem)[0];
+    picker.on('emoji', emoji => {
+      $(iconElem).attr('emoji', emoji).html(twemoji.parse(emoji));
+    });
+    $(icon).off().click(() => {
```
Comment (okriuchykhin) [inline app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js]: Try to use event delegation for this

File: app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js [hunk 2]
```diff
@@ -47,14 +48,56 @@ var RepositoryStatusColumnType = (function() {
...
+      if (picker.pickerVisible) {
+        picker.hidePicker();
+      } else {
+        picker.showPicker(icon);
+      }
+      setTimeout(() => {
+        replaceEmojies($('.emoji-picker.visible .emoji-picker__tab-body.active'));
+        $.each($('.emoji-picker__tab-body .emoji-picker__emojis'), (i, scrollContainer) => {
+          $(scrollContainer).off('scroll').on('scroll', function() {
```
Comment (okriuchykhin) [inline app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js]: Use event delegation here

File: app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js [hunk 3]
```diff
@@ -47,14 +48,56 @@ var RepositoryStatusColumnType = (function() {
...
+            replaceEmojies($(this).closest('.emoji-picker__tab-body'));
+          });
+        });
+      }, 100);
+      $('.emoji-picker__tabs').off('click', '.emoji-picker__tab')
```
Comment (okriuchykhin) [inline app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js]: Use event delegation here

File: app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js [hunk 4]
```diff
@@ -47,14 +48,56 @@ var RepositoryStatusColumnType = (function() {
...
+        .on('click', '.emoji-picker__tab', function() {
+          $.each($('.emoji-picker__tab'), (i, tab) => {
+            if ($(tab).hasClass('active')) {
+              replaceEmojies($($('.emoji-picker__tab-body')[i]));
+            }
+          });
+        });
+
+      $('.emoji-picker__tab-body.active').off('click', '.emoji-picker__emoji')
```
Comment (okriuchykhin) [inline app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js]: Use event delegation here

File: app/assets/stylesheets/repository/repository_table.scss [hunk 5]
```diff
@@ -94,6 +90,47 @@
...
+  .repository-status-value-icon {
+    .emoji {
+      height: 24px;
+      margin-right: 5px;
+      min-width: 24px;
+      width: 24px;
+    }
+  }
+
+  .dropdown-selector-container {
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/assets/stylesheets/repository/repository_table.scss]: Fix this

[the remark this answers is not in the document]
Comment (aignatov-bio) [reply app/assets/stylesheets/repository/repository_table.scss]: Fixed

File: app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js [hunk 6]
```diff
@@ -46,17 +47,35 @@ var RepositoryStatusColumnType = (function() {
...
-      autocomplete: false,
-      pickerPosition: 'bottom',
-      events: {
-        emojibtn_click: function(button, event) {
-          validateForm();
+  function initEmojiPicker() {
+    // init Emoji picker modal
+    $(manageModal)
+      .on('click', iconId, function() {
+        var picker = new EmojiButton({ rootElement: document.getElementById('manage-repository-column') });
```
Comment (okriuchykhin) [inline app/assets/javascripts/repository_columns/columns_initializers/status_column_type.js]: put manageModal instead 'manage-repository-column'

**Pull request merged**